extensions/functions_datetime.yaml

```diff
@@ -6,30 +6,26 @@ scalar_functions:
     description: Extract portion of a date/time value.
     impls:
       - args:
-          - options: [ YEAR, MONTH, DAY, SECOND ]
+          - values: [ YEAR, MONTH, DAY, SECOND ]
             name: The part of the value to extract.
-            required: true
           - name: x
             value: timestamp
         return: i64
       - args:
-          - options: [ YEAR, MONTH, DAY, SECOND ]
+          - values: [ YEAR, MONTH, DAY, SECOND ]
```

Comment:
I'm surprised by your change here. Are you renaming required enumeration argument types?

Reply:
My choices were to leave this alone and have it be out of sync with options (`values` vs `options`) or to have the word "options" doubly used in options `options::options`.  However, if `options` is a map then this is a moot point so I will revert this change.

Reply:
Actually I take that back.  The map change doesn't fix this.  So basically we can have approach 1:

```
args:
 - name: some_required_enum
   options: [ X, Y ]
options:
 some_optional_enum:
  options: [ A, B ]
```

Which is consistent but has some shadowing with the double options.  Or we can have approach 2:

```
args:
 - name: some_required_enum
   options: [ X, Y ]
options:
 some_optional_enum:
  values: [ A, B ]
```

This avoids the shadowing but is inconsistent (options vs. values).  Or we can have approach 3:

```
args:
 - name: some_required_enum
   values: [ X, Y ]
options:
 some_optional_enum:
   values: [ A, B ]
```

This avoids the shadowing and is consistent but is more of a breaking change.

I'm happy to go with any of the above choices.

Reply:
I have currently implemented approach 2.